app/code/Transoft/Blog/Controller/Adminhtml/Post/Save.php

```diff
@@ -0,0 +1,144 @@
+    private $modelRepository;
+
+    /**
+     * Constructor
+     *
+     * @param Context $context
+     * @param PageFactory $resultPageFactory
+     * @param DataPersistorInterface $dataPersistor
+     * @param ModelFactory $modelFactory
+     * @param ModelRepositoryInterface $modelRepository
+     */
+    public function __construct(
+        Context $context,
+        PageFactory $resultPageFactory,
+        DataPersistorInterface $dataPersistor,
+        ModelFactory $modelFactory = null,
+        ModelRepositoryInterface $modelRepository = null
+    ) {
+        $this->resultPageFactory = $resultPageFactory;
+        $this->dataPersistor = $dataPersistor;
+        $this->modelFactory = $modelFactory
+            ?: ObjectManager::getInstance()->get(ModelFactory::class);
+        $this->modelRepository = $modelRepository
+            ?: ObjectManager::getInstance()->get(ModelRepositoryInterface::class);
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+
+        if ($data) {
+            if (empty($data['blog_id'])) {
+                $data['blog_id'] = null;
+            }
+
+            $post = $this->modelFactory->create();
+
+            $id = $this->getRequest()->getParam('id');
+            $image_url = $data['image_path'][0]['url'];
+            $data['image_path'] = $image_url;
+
+            if ($id) {
+                try {
+                    $post = $this->modelRepository->getById($id);
+                } catch (LocalizedException $e) {
+                    $this->messageManager->addErrorMessage(__('This post no longer exists.'));
+                    return $resultRedirect->setPath('*/*/');
+                }
+            }
+
+            try {
+                $this->modelRepository->save($post);
+                $this->messageManager->addSuccessMessage(__('You saved the post.'));
+                $this->dataPersistor->clear('transoft_blog');
+                return $this->processBlockReturn($post, $data, $resultRedirect);
+            } catch (LocalizedException $e) {
+                $this->messageManager->addErrorMessage($e->getMessage());
+            } catch (\Exception $e) {
+                $this->messageManager->addExceptionMessage($e, __('Something went wrong while saving the post.'));
+            }
+
+            $this->dataPersistor->set('transoft_blog', $data);
+            return $resultRedirect->setPath('*/*/edit', ['blog_id' => $id]);
+        }
+        return $resultRedirect->setPath('*/*/');
+    }
+
+    /**
+     * Process and set the block return
+     *
+     * @param Model $model
+     * @param array $data
+     * @param ResultInterface $resultRedirect
+     * @return ResultInterface
+     * @throws CouldNotSaveException
+     */
+    private function processBlockReturn($model, $data, $resultRedirect)
```

Comment:
Provide type hinting and method return type.

Reply:
provided